src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,374 @@
+            </View>
+          </View>
+        </ScrollView>
+      )}
+      <Button
+        style={{ padding: Spacing.Regular16 }}
+        onPress={() => Logger.debug('TODO Perform Swap!!!')}
+        text={t('swapReviewScreen.complete')}
+        size={BtnSizes.FULL}
+        disabled={loading || fetchSwapQuoteError}
+      />
+      {/** Estimated amount dialog */}
+      <Dialog
+        title={t('swapReviewScreen.estimatedAmountTitle')}
+        isVisible={estimatedModalVisible}
+        actionText={t('dismiss')}
+        actionPress={() => setEstimatedDialogVisible(false)}
+        isActionHighlighted={false}
+        onBackgroundPress={() => setEstimatedDialogVisible(false)}
+      >
+        {t('swapReviewScreen.estimatedAmountBody', {
+          slippagePercent: maxSlippagePercent,
+        })}
+      </Dialog>
+      {/** App fee dialog */}
+      <Dialog
+        title={t('swapReviewScreen.swapFeeTitle')}
+        isVisible={swapFeeModalVisible}
+        actionText={t('dismiss')}
+        actionPress={() => setSwapFeeModalVisible(false)}
+        isActionHighlighted={false}
+        onBackgroundPress={() => setSwapFeeModalVisible(false)}
+      >
+        {t('swapReviewScreen.swapFeeBody', {
+          swapFee: swapFeePercentage,
+        })}
+      </Dialog>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  safeAreaContainer: {
+    flex: 1,
+  },
+  loadingContentContainer: {
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  contentContainer: {
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  subContentContainer: {
+    paddingBottom: Spacing.Regular16,
+  },
+  tallRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Regular16,
+  },
+  row: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Smallest8,
+  },
+  tokenDisplayView: {
+    flex: 1,
+    justifyContent: 'space-between',
+  },
+  separator: {
+    height: 1,
+    width: '100%',
+    backgroundColor: colors.gray2,
+  },
+  transactionDetailsRightText: {
+    ...fontStyles.regular,
+    flex: 1,
+    alignSelf: 'flex-end',
+    textAlign: 'right',
+    textAlignVertical: 'center',
+    fontWeight: '400',
+    lineHeight: 24,
+  },
+  touchableRow: {
+    alignSelf: 'flex-end',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  transactionDetailsLabel: {
+    ...fontStyles.label,
+    paddingBottom: Spacing.Smallest8,
+    color: colors.gray4,
+  },
+  label: {
+    flex: 1,
+    ...fontStyles.regular,
+  },
+  amountText: {
+    textAlign: 'right',
+    ...fontStyles.large,
+  },
+  amountSubText: {
+    textAlign: 'right',
+    ...fontStyles.small,
+    color: colors.gray4,
+  },
+  feeWaived: {
+    textDecorationLine: 'line-through',
+    textDecorationStyle: 'solid',
+  },
+})
+
+SwapReviewScreen.navOptions = {
```

Comment:
do we need this?

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2767/commits/a4c7a335213039b2faef30dea75fa44f1b427aa5!

Reply:
ah, i was confused about why the SwapScreen doesn't need this but turns out the swap screen is a drawer screen. all good!